src/com/trollworks/gcs/widgets/outline/MultiCell.java

```diff
@@ -109,9 +111,35 @@ protected String getSortText(ListRow row) {
      */
     @SuppressWarnings("static-method")
     protected String getSecondaryText(ListRow row) {
-        String modifierNotes = row.getModifierNotes();
-        String notes         = row.getNotes();
-        return modifierNotes.length() == 0 ? notes : modifierNotes + '\n' + notes;
+        StringBuilder builder = new StringBuilder();
+        if (DisplayPreferences.showUserDescInDisplay()) {
+            if (row instanceof Advantage) {
+                Advantage ad   = (Advantage) row;
+                String    desc = ad.getUserDesc();
+                builder.append(desc);
+                if (desc.length() > 0) {
+                    builder.append('\n');
+                }
+            }
+        }
+        if (DisplayPreferences.showNotesInDisplay()) {
+            String desc = row.getNotes();
+            builder.append(desc);
+            if (desc.length() > 0) {
+                builder.append('\n');
+            }
+        }
+        if (DisplayPreferences.showModifiersInDisplay()) {
+            String desc = row.getModifierNotes();
+            builder.append(desc);
+            if (desc.length() > 0) {
+                builder.append('\n');
+            }
+        }
+        if (builder.length() > 0) {
+            builder.setLength(builder.length() - 1);   // Remove the last '\n'
+        }
+        return builder.toString();
```

Comment:
This code seems very similar to the code up above. Would probably be worthwhile to extract a single method for doing this and use it in both places.

Reply:
I have refactored what I could, but there are significant differences.   One is checking the "*InDisplay" preferences, and the other is checking the "AsToolTip" preferences.